Simplenote/Classes/NoteOptionsViewController.swift

```diff
@@ -241,42 +292,177 @@ class NoteOptionsViewController: UITableViewController {
     // MARK: - Row Action Handling
     @objc
     func handlePinToTop(sender: UISwitch) {
-        ///Handle pinning logic here
+        note.pinned = sender.isOn
+        save()
+
+        sender.accessibilityHint = note.pinned ?
+            NSLocalizedString("Unpin note", comment: "Action to mark a note as unpinned") :
+            NSLocalizedString("Pin note", comment: "Action to mark a note as pinned")
+
+        if sender.isOn {
+            SPTracker.trackEditorNotePinned()
+        } else {
+            SPTracker.trackEditorNoteUnpinned()
+        }
     }
```

Comment:
There's a bit of code duplication in all of the event handlers. Mind taking a peek sir?

Reply:
Thanks, tidied this up now to stop duplication of if statements and moved the presentation code into a new method.

Reply:
Thank you @mattdc28 !! Just one last note. The accessibilityHint assignment itself is duplicated (so the strings are x2, and it's slightly harder to maintain).

Same note for the other API(s)!